composer/trainer/trainer.py

```diff
@@ -756,6 +774,8 @@ def __init__(
         self.state = State(
             rank_zero_seed=rank_zero_seed,
             algorithms=algorithms,
+            train_dataloader=train_dataloader.dataloader,
```

Comment:
Instead of binding the the `train_dataloader` and `evaluators` to the `State` before `Event.INIT`, would it be possible set these attributes after this event?

https://github.com/mosaicml/composer/blob/23f888e07456d31a2950db9f962f4dbdf41938fb/composer/trainer/trainer.py#L899-L901

The rational is it will be valid to pass it data loaders and evaluators on either `Trainer.__init__` or `Trainer.fit()`, so code should (algorithms and callbacks) should not depend on them being specified on `Event.INIT`.

Reply:
Can these two lines be removed? I see they are being set below.